adafruit_framebuf.py

```diff
@@ -36,6 +37,60 @@
 GS4_HMSB = 2  # Unimplemented!
 MHMSB = 3  # Single bit displays like the Sharp Memory
 RGB888 = 4  # Neopixels and Dotstars
+GS2_HMSB = 5  # 2-bit color displays like the HT16K33 8x8 Matrix
+
+
+class GS2HMSBFormat:
+    """GS2HMSBFormat"""
+
+    @staticmethod
+    def set_pixel(framebuf, x, y, color):
+        """Set a given pixel to a color."""
+        index = (y * framebuf.stride + x) >> 2
+        pixel = framebuf.buf[index]
+
+        shift = (x & 0b11) << 1
+        mask = 0b11 << shift
+        color = (color & 0b11) << shift
```

Comment:
I don't actually understand why we right shift 2, then left shift 1 and then left shift by `shift`. I wonder if there is a conceptually simpler way to write this.